packages/web3-providers/src/resolvers/ProviderResolver.js

```diff
@@ -82,7 +82,7 @@ export default class ProviderResolver {
             return this.providersModuleFactory.createMistEthereumProvider(provider);
         }
 
-        if (provider.isEIP1193) {
+        if (typeof provider.send === 'function' && typeof provider.on === 'function') {
```

Comment:
Could you put this ``if`` statement below the metamask provider check? This because the metamask provider does have a send and on method as well. I will test it after with the ``web3-provider-engine`` and the ``truffle-hdwallet-provider``

Reply:
fixed: https://github.com/ethereum/web3.js/pull/2922/commits/1f8bb4e22bc849b9d48df314650a29438a15ab80